File: app/src/main/java/com/xsolla/android/storesdkexample/fragments/ResetPasswordFragment.java
package com.xsolla.android.storesdkexample.fragments;

import android.widget.Button;
import android.widget.TextView;

import com.xsolla.android.login.XLogin;
import com.xsolla.android.login.callback.ResetPasswordCallback;
import com.xsolla.android.storesdkexample.R;
import com.xsolla.android.storesdkexample.fragments.base.BaseFragment;
import com.xsolla.android.storesdkexample.util.ViewUtils;

public class ResetPasswordFragment extends BaseFragment {

    private TextView usernameInput;
    private Button resetPasswordButton;

    @Override
    public int getLayout() {
        return R.layout.fragment_reset_password;
    }

    @Override
    public void initUI() {
        usernameInput = rootView.findViewById(R.id.username_input);
        resetPasswordButton = rootView.findViewById(R.id.reset_password_button);
        resetPasswordButton.setOnClickListener(v -> {
            ViewUtils.disable(resetPasswordButton);
            resetPassword();
        });
    }

    private void resetPassword() {
        hideKeyboard();
        String username = usernameInput.getText().toString();
        XLogin.resetPassword(username, new ResetPasswordCallback() {
            @Override
            public void onSuccess() {
                showSnack("Password reset success. Check your email");
                openRootFragment();
                ViewUtils.enable(resetPasswordButton);
            }

            @Override
            public void onError(Throwable throwable, String errorMessage) {
                if (errorMessage != null) {
                    showSnack(errorMessage);
                } else {
                    showSnack(throwable.getClass().getName());
                }
                ViewUtils.enable(resetPasswordButton);
            }
        });
    }

}

File: app/src/main/java/com/xsolla/android/storesdkexample/fragments/ResetPasswordFragment.kt
package com.xsolla.android.storesdkexample.fragments

import com.xsolla.android.login.XLogin
import com.xsolla.android.login.api.XLoginCallback
import com.xsolla.android.storesdkexample.R
import com.xsolla.android.storesdkexample.fragments.base.BaseFragment
import com.xsolla.android.storesdkexample.util.ViewUtils
import kotlinx.android.synthetic.main.fragmen_reset_password.view.*

class ResetPasswordFragment : BaseFragment() {

    override fun getLayout(): Int {
        return R.layout.fragmen_reset_password
    }

    override fun initUI() {
        rootView.toolbar.setNavigationIcon(R.drawable.ic_arrow_back_white_24dp)
        rootView.toolbar.setNavigationOnClickListener { popFragment() }

        rootView.resetPasswordButton.setOnClickListener { v ->
            ViewUtils.disable(v)
            hideKeyboard()
            val username = rootView.usernameInput.text.toString()
            XLogin.resetPassword(username, object : XLoginCallback<Void?>() {
                override fun onSuccess(response: Void?) {
                    showSnack("Password reset success. Check your email")
                    openLoginFragment()
                    ViewUtils.enable(v)
                }

                override fun onFailure(errorMessage: String) {
                    showSnack(errorMessage)
                    ViewUtils.enable(v)
                }
            })
        }
    }

    private fun openLoginFragment() {
        activity?.let {
            it.supportFragmentManager.popBackStack()
        }
    }
}